sound/soc/soc-topology.c

```diff
@@ -910,82 +910,29 @@ static int soc_tplg_dbytes_create(struct soc_tplg *tplg, size_t size)
 
 static int soc_tplg_dmixer_create(struct soc_tplg *tplg, size_t size)
 {
-	struct snd_soc_tplg_mixer_control *mc;
-	struct soc_mixer_control *sm;
-	struct snd_kcontrol_new kc;
-	int ret = 0;
-
-	if (soc_tplg_check_elem_count(tplg,
-				      sizeof(struct snd_soc_tplg_mixer_control),
-				      1, size, "mixers"))
-		return -EINVAL;
-
-	mc = (struct snd_soc_tplg_mixer_control *)tplg->pos;
-
-	/* validate kcontrol */
-	if (strnlen(mc->hdr.name, SNDRV_CTL_ELEM_ID_NAME_MAXLEN) ==
-		SNDRV_CTL_ELEM_ID_NAME_MAXLEN)
-		return -EINVAL;
-
-	sm = devm_kzalloc(tplg->dev, sizeof(*sm), GFP_KERNEL);
-	if (sm == NULL)
-		return -ENOMEM;
-	tplg->pos += (sizeof(struct snd_soc_tplg_mixer_control) +
-		      le32_to_cpu(mc->priv.size));
-
-	dev_dbg(tplg->dev,
-		"ASoC: adding mixer kcontrol %s with access 0x%x\n",
-		mc->hdr.name, mc->hdr.access);
-
-	memset(&kc, 0, sizeof(kc));
-	kc.name = mc->hdr.name;
-	kc.private_value = (long)sm;
-	kc.iface = SNDRV_CTL_ELEM_IFACE_MIXER;
-	kc.access = le32_to_cpu(mc->hdr.access);
+	struct snd_kcontrol_new kc = {0};
+	struct soc_bytes_ext *sbe;
```

Comment:
this one I am not following at all. There was no 'bytes_ext' in the existing code?

Reply:
Eh... a whole patch is a mess, as it should replace the code with mixer kcontrol (not bytes), will check it again tomorrow with fresh mind, to make sure I didn't miss something else.